### PR TITLE
Refact: clean code XCTSkip 

### PR DESCRIPTION
**Pull request by jayahariv**
* keeps forgetting, whether the test skips or really passing. 
* XCTSkip can be really helpful in detecting this case

### CODE REVIEW

**Comment by pasin** [comment]
This feature doesn't support by XCode 10 so we can't build the test to run on AWS device farm. Can this be hold off?

File: Swift/Tests/TLSIdentityTest.swift [hunk 1]
```diff
@@ -149,7 +149,7 @@ class TLSIdentityTest: CBLTestCase {
     }
     
     func testCreateGetDeleteServerIdentity() throws {
-        if (!keyChainAccessAllowed) { return }
+        try XCTSkipUnless(keyChainAccessAllowed, "Skipping, keychain is not allowed in this config")
```
Comment (pasin) [inline Swift/Tests/TLSIdentityTest.swift:152]: Don't we need to ignore exception here?

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 2]
```diff
@@ -873,7 +901,9 @@ - (void) _testReadOnlyListener {
 }
 
 - (void) testCloseWithActiveListener {
-    if (!self.keyChainAccessAllowed) return;
+    [self ignoreExceptionBreakPointOnly: ^{
+        XCTSkipUnless(self.keyChainAccessAllowed, @"Skipping, keychain not accessible in this config");
+    }];
```
Comment (pasin) [inline Objective-C/Tests/URLEndpointListenerTest.m:906]: Can we extract this into a method to make this cleaner?

File: Swift/Tests/TLSIdentityTest.swift (hunk 1 above)
Comment (jayahariv) [reply Swift/Tests/TLSIdentityTest.swift:152]: Exception pointer is only hit on ObjC, and not in Swift.

Comment (pasin) [reply Swift/Tests/TLSIdentityTest.swift:152]: I don't know. Can you wrap this in a function such as checkKeyChainAccess()?

**Comment by jayahariv** [comment]
once device farm is ready, please merge.

**Pull request closed without merging**